File: translated/f-to-c/DRB102-copyprivate-orig-no.c
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.

SPDX-License-Identifier: (BSD-3-Clause)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//threadprivate+copyprivate: no data races

#include <omp.h>
#include <stdio.h>

// Global variables (from module)
int y;
double x;

#pragma omp threadprivate(x, y)

int main()
{
    #pragma omp parallel
    {
        #pragma omp single copyprivate(x, y)
        {
            x = 1.0;
            y = 1;
        }
    }

    printf("x = %.1f  y = %d\n", x, y);

    return 0;
} 

File: translated/f-to-go-automodel-bad/DRB102-copyprivate-orig-no.go
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
// and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
//
// SPDX-License-Identifier: (BSD-3-Clause)
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

//threadprivate+copyprivate: no data races

package main

import (
	"fmt"
	"sync"
)

// Module DRB102 translated to package-level variables
var y int
var x float64

func main() {
	//$omp parallel
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		//$omp single
		x = 1.0
		y = 1
		//$omp end single copyprivate(x,y)
	}()
	wg.Wait()
	//$omp end parallel

	fmt.Printf("x =%3.1f  y =%3d\n", x, y)
}
